<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="mplWidget" name="mplWidget" native="true"/>
    </item>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout">
      <property name="rightMargin">
       <number>10</number>
      </property>
      <item>
       <spacer name="vertical_spacer_top">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <layout class="QHBoxLayout" name="grid_width_layout">
        <item>
         <widget class="QLabel" name="grid_width_label">
          <property name="statusTip">
           <string/>
          </property>
          <property name="whatsThis">
           <string/>
          </property>
          <property name="text">
           <string>Width:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="grid_width_slider_2">
          <property name="statusTip">
           <string>Width of game</string>
          </property>
          <property name="whatsThis">
           <string>Width of game</string>
          </property>
          <property name="maximum">
           <number>500</number>
          </property>
          <property name="value">
           <number>20</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSlider" name="grid_width_slider">
          <property name="statusTip">
           <string>Width of game</string>
          </property>
          <property name="whatsThis">
           <string>Width of game</string>
          </property>
          <property name="maximum">
           <number>200</number>
          </property>
          <property name="value">
           <number>20</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="grid_height_layout">
        <item>
         <widget class="QLabel" name="grid_height_label">
          <property name="statusTip">
           <string/>
          </property>
          <property name="text">
           <string>Height</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="grid_height_spinbox">
          <property name="statusTip">
           <string>Height of game</string>
          </property>
          <property name="whatsThis">
           <string>Height of game</string>
          </property>
          <property name="maximum">
           <number>500</number>
          </property>
          <property name="value">
           <number>20</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSlider" name="grid_height_slider">
          <property name="statusTip">
           <string>Height of game</string>
          </property>
          <property name="whatsThis">
           <string>Height of game</string>
          </property>
          <property name="maximum">
           <number>200</number>
          </property>
          <property name="value">
           <number>20</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="speed_layout">
        <property name="spacing">
         <number>6</number>
        </property>
        <property name="sizeConstraint">
         <enum>QLayout::SetDefaultConstraint</enum>
        </property>
        <item>
         <widget class="QLabel" name="speed_label">
          <property name="toolTip">
           <string>Speed</string>
          </property>
          <property name="statusTip">
           <string>Number of generations per second</string>
          </property>
          <property name="whatsThis">
           <string>Number of generations per second</string>
          </property>
          <property name="text">
           <string>Speed:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="speed_spin_box">
          <property name="toolTip">
           <string>Speed</string>
          </property>
          <property name="statusTip">
           <string>Number of generations per second</string>
          </property>
          <property name="whatsThis">
           <string>Number of generations per second</string>
          </property>
          <property name="maximum">
           <number>999</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSlider" name="speed_slider">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="play_next_layout">
        <item>
         <widget class="QPushButton" name="iteration_play">
          <property name="maximumSize">
           <size>
            <width>30</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="icon">
           <iconset>
            <normaloff>../assets/baseline_play_arrow_black_18dp.png</normaloff>../assets/baseline_play_arrow_black_18dp.png</iconset>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
          <property name="autoRepeat">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="iteration_next">
          <property name="maximumSize">
           <size>
            <width>30</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="icon">
           <iconset>
            <normaloff>../assets/baseline_skip_next_black_18dp.png</normaloff>../assets/baseline_skip_next_black_18dp.png</iconset>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QPushButton" name="restart_game">
        <property name="text">
         <string>Restart Game</string>
        </property>
        <property name="checkable">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="vertical_space_bottom">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>mplWidget</class>
   <extends>QWidget</extends>
   <header>mplwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../assets/source.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>speed_slider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>speed_spin_box</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>555</x>
     <y>300</y>
    </hint>
    <hint type="destinationlabel">
     <x>471</x>
     <y>300</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>grid_height_slider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>grid_height_spinbox</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>554</x>
     <y>276</y>
    </hint>
    <hint type="destinationlabel">
     <x>452</x>
     <y>275</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>grid_width_slider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>grid_width_slider_2</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>538</x>
     <y>239</y>
    </hint>
    <hint type="destinationlabel">
     <x>455</x>
     <y>238</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
